src/lib/y2storage/proposal_space_settings.rb

```diff
@@ -0,0 +1,100 @@
+# Copyright (c) [2023] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "y2storage/equal_by_instance_variables"
+
+module Y2Storage
+  # Class to encapsulate all the GuidedProposal settings related to the process of making space
+  # to allocate the new operating system
+  class ProposalSpaceSettings
+    include EqualByInstanceVariables
```

Comment:
Just as note, we have the `Equatable` mixin for custom comparison: https://github.com/yast/yast-yast2/blob/master/library/general/src/lib/yast2/equatable.rb

Reply:
Yes, only mentioning to promote it for other cases :)